tools/deployment/linux_check_init.sh

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+
+#  Copyright (c) 2014-present, Facebook, Inc.
+#  All rights reserved.
+#
+#  This source code is licensed under the BSD-style license found in the
+#  LICENSE file in the root directory of this source tree. An additional grant
+#  of patent rights can be found in the PATENTS file in the same directory.
+
+SYSTEMD_SERVICE_DIR="/usr/lib/systemd/system"
+SYSTEMD_SYSCONFIG_DIR="/etc/sysconfig"
+SYSTEMD_SYSCONFIG_DIR_DEB="/etc/default"
+INITD_DIR="/etc/rc.d/init.d"
+INITD_DIR_DEB="/etc/init.d"
+
+# Check if OS is Debian based
+if ls /etc | grep -Fq debian_version; then
+  SYSTEMD_SYSCONFIG_DIR=$SYSTEMD_SYSCONFIG_DIR_DEB
+  INITD_DIR=$INITD_DIR_DEB
+fi
+
+# Check if system is using upstart
+if ps -p 1 | grep -Fq init && init --version | grep -Fq upstart; then
+  rm $SYSTEMD_SYSCONFIG_DIR/osqueryd
```

Comment:
So instead of removing files that are part of the package, could this post-install script just move content into the expected location (from a `/usr/share/osquery/systemd/` folder or similar)? Then at uninstall a script could remove those files created by the post-install described above?

If that's possible, would it then avoid the CentOS/RPM uninstall warning?